source/github-events/on-new-comments.ts

```diff
@@ -50,20 +50,18 @@ function addListeners(): delegate.Subscription[] {
 		childList: true,
 	});
 
-	return [
-		// When hidden comments are loaded by clicking "Load more…"
-		delegate(document, '.js-ajax-pagination', 'submit', paginationSubmitHandler),
+	// When hidden comments are loaded by clicking "Load more…"
+	delegate(document, '.js-ajax-pagination', 'submit', paginationSubmitHandler, {signal});
 
-		// Collapsed comments are loaded later using an include-fragment element
-		delegate(document, 'details.js-comment-container include-fragment:not([class])', 'loadstart', getFragmentLoadHandler(run), true),
-	];
+	// Collapsed comments are loaded later using an include-fragment element
+	delegate(document, 'details.js-comment-container include-fragment:not([class])', 'loadstart', getFragmentLoadHandler(run), {capture: true, signal});
 }
 
-export default function onNewComments(callback: VoidFunction): Deinit {
+export default function onNewComments(callback: VoidFunction, signal: AbortSignal): Deinit {
+	addListeners(signal);
 	handlers.add(callback);
 
 	return [
-		...addListeners(),
 		handlers.clear,
 		observer,
 	];
```

Comment:
If it takes a signal, it should use it for everything 

```js
onAbort(signal, handlers, observer);// It could accept multiple objects, like the `getDeinit` function can handle it all
```